### PR TITLE
HHH-6686 Fix JQL exception in face of 'is (not) empty'

### PR DESCRIPTION
**Pull request by NathanQingyangXu**
https://hibernate.atlassian.net/browse/HHH-6686

This 'ancient' bug aroused my attention. However, the fixing seems straightforward.
Testing case attached.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/hql/internal/ast/HqlParser.java [hunk 1]
```diff
@@ -348,6 +347,10 @@ private AST processIsEmpty(AST node, boolean negated) {
 	private AST createSubquery(AST node) {
 		AST ast = ASTUtil.createParent( astFactory, RANGE, "RANGE", node );
 		ast = ASTUtil.createParent( astFactory, FROM, "from", ast );
+
+		AST alias = ASTUtil.createSibling( astFactory, ALIAS, "_", node );
+		ASTUtil.insertChild( ASTUtil.createSibling( astFactory, SELECT, "select", ast ), astFactory.create(IDENT, alias.getText() ) );
+
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/hql/internal/ast/HqlParser.java]: The root cause of the bug is for 'exists' sub-query, we didn't specify 'SELECT', though we specified 'FROM'. Adding 'SELECT' fixed the issue.

File: hibernate-core/src/main/java/org/hibernate/hql/internal/ast/HqlParser.java [hunk 2]
```diff
@@ -512,11 +515,11 @@ private String toPathText(AST node) {
...
 	public Map<String, Set<String>> getTreatMap() {
-		return treatMap == null ? Collections.<String, Set<String>>emptyMap() : treatMap;
+		return treatMap == null ? Collections.emptyMap() : treatMap;
 	}
 
 	public static void panic() {
-		//overriden to avoid System.exit
+		//overridden to avoid System.exit
 		throw new QueryException( "Parser: panic" );
 	}
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/hql/internal/ast/HqlParser.java]: I tried to fix some cosmetic legacy issues along the way. Not sure it is a good idea though.

**Comment by dreab8** [comment]
applied upstream.

Thanks @NathanQingyangXu

**Comment by sebersole** [comment]
Nice!

**Comment by Benschio** [comment]
Very nice, thank you very much!

**Pull request closed without merging**